pkg/doc/vc/crypto/crypto.go

```diff
@@ -188,7 +189,7 @@ type Crypto struct {
 }
 
 // SignCredential signs vc.
-func (c *Crypto) SignCredential(dataProfile *vcprofile.DataProfile, vc *verifiable.Credential,
+func (c *Crypto) SignCredential(dataProfile *vcsstorage.DataProfile, vc *verifiable.Credential,
```

Comment:
Do we need to put everything in a storage package? Can't we modularize storage for each domain the way it is? Like issuer, verifier, holder, csl, crypto?

Reply:
I was looking into this as well. Can we minimize changes here and make refactoring the next task as part of implementing new API (we have a better context on how it is going to be used). cc @DRK3, @rolsonquadras

Reply:
@rolsonquadras I could move the interface and data type definitions into separate packages, or perhaps move them into their respective operations files. (Part of the reason for moving the data type/models into the storage folder was to avoid circular dependencies with the original way I implemented it in my original iteration of the PR, but now that's not an issue)

However, on a similar note as what @aholovko said, I think it would be more productive to not rework this PR too many times considering the large refactor work that @aholovko and @vkubiv are working on. I'd suggest merging this in as-is and then I'll work with @aholovko and @vkubiv to refactor the storage code in a better way that also works with the new design. The most important part of this PR is to get the MongoDB storage optimizations in.